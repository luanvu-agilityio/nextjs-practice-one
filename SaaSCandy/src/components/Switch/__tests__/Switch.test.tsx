import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Switch } from '../index';

describe('Switch Component', () => {
  it('matches snapshot', () => {
    const { container } = render(<Switch />);
    expect(container).toMatchSnapshot();
  });

  it('toggles state when clicked', async () => {
    const user = userEvent.setup();
    const handleChange = jest.fn();
    const { getByRole } = render(<Switch onCheckedChange={handleChange} />);

    const switchElement = getByRole('switch');
    await user.click(switchElement);

    expect(handleChange).toHaveBeenCalledWith(true);
  });
});
