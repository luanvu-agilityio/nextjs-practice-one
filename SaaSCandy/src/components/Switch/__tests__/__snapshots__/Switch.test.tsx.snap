// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Switch Component matches snapshot 1`] = `
<div>
  <button
    aria-checked="false"
    class="peer inline-flex h-7 w-14 shrink-0 cursor-pointer items-center rounded-full border-2 border-gray-900 transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 focus-visible:ring-offset-background disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-gray-900 data-[state=unchecked]:bg-white"
    data-state="unchecked"
    role="switch"
    type="button"
    value="on"
  >
    <span
      class="pointer-events-none block h-4 w-4 rounded-full bg-gray-900 shadow-lg ring-0 transition-transform data-[state=checked]:translate-x-7 data-[state=checked]:bg-white data-[state=unchecked]:translate-x-0.5"
      data-state="unchecked"
    />
  </button>
</div>
`;
